pure.zsh

```diff
@@ -57,17 +59,28 @@ prompt_pure_clear_screen() {
 }
 
 prompt_pure_check_git_arrows() {
+	# reset git arrows
+	prompt_pure_git_arrows=
+
 	# check if there is an upstream configured for this branch
 	command git rev-parse --abbrev-ref @'{u}' &>/dev/null || return
 
-	local right left arrows
-	right=$(command git rev-list --right-only --count HEAD...@'{u}' 2>/dev/null)
-	left=$(command git rev-list --left-only --count HEAD...@'{u}' 2>/dev/null)
+	# check arrow status inside single subshell for performance
+	local output
+	output="$(
```

Comment:
You can but declaring and assigning in one statement masks the exit code, example:

```zsh
local output=$(exit 1)
echo $? # 0

output=$(exit 1)
echo $? # 1
```